### DISCUSSION TITLE
Dockerfile to generate a SDK Image With Wix 4

### DISCUSSION THREAD

**Discussion by opuech** [wixtoolset/issues#7972] [Questions] [answered]
Based on these Dockerfiles
- SDK 4.7.2 : https://github.com/microsoft/dotnet-framework-docker/blob/062715aec03a503f9e005203adddb7231c855a1b/4.7.2/sdk/windowsservercore-ltsc2019/Dockerfile
- SDK 6 : https://github.com/dotnet/dotnet-docker/blob/main/src/sdk/6.0/windowsservercore-ltsc2022/amd64/Dockerfile

And this documentation : https://github.com/wixtoolset/wix

> Prerequisites
> A command line Git client that is in the system path
> Visual Studio 2022 (17.8.2 or higher) with the following installed:
> Workloads
> ASP.NET and web development
> .NET desktop development
> Desktop development with C++
> Individual components
> .NET 6.0 Runtime (Long Term Support)
> .NET Framework 4.7.2 SDK
> .NET Framework 4.7.2 targeting pack
> .NET Framework 4.6.2 targeting pack
> MSVC v141 - VS 2017 C++ ARM64 build tools (v14.16)
> MSVC v141 - VS 2017 C++ x64/x86 build tools (v14.16)
> MSVC v143 - VS 2022 C++ ARM64/ARM64EC build tools (Latest)
> MSVC v143 - VS 2022 C++ x64/x86 build tools (Latest)

How to install in command line this workloads ?
- Microsoft.VisualStudio.Product.BuildTools https://learn.microsoft.com/en-us/visualstudio/install/workload-and-component-ids?view=vs-2022
  (Seems already done)
- Microsoft.VisualStudio.Workload.VCTools https://learn.microsoft.com/en-us/visualstudio/install/workload-component-id-vs-build-tools?view=vs-2022&preserve-view=true (seems to be missing)

Could you help me to create a Dockerfile that generate a SDK Image With Wix 4 in order to execute msbuild on a wix project sucessfully ?

**Comment by chrpai** [comment, marked as answer]
I'm not sure why you would need any of this.  WiX v4 is dotnet sdk / nuget based and will automatically restore the dependencies as part of your build.  Just having the base docker image should be enough.

**Reply by opuech** [reply]
Thank you ! (Some missing dependencies on my project generated some confusing errors). 

If it is full "dotnet sdk / nuget based", it should work the same way on Linux or Windows containers. Is it right ?

**Reply by chrpai** [reply]
As WiX makes Windows Installer packages and ultimately needs MSI.dll, WiX only runs on Windows.  Theoretically, you could run it on Linux with Wine but that's going to be a lot of upstream swimming and YMMV.

**Reply by opuech** [reply]
I confirm that it works on windows container by using simple sdk image. 
On Linux, i have this error that confirms the incompatibility :
> wix.exe : warning WIX0000: The WiX Toolset only supports Windows. If you would like to help bring WiX to other platforms, join us at https://wixtoolset.org. All behavior after this point is undefined.
> error WIX0389: The Directory/@Name attribute's value, '!(bind.Property.Manufacturer) !(bind.Property.ProductName)', is not a relative path. [/src/InstallKit/DesktopApplication/DesktopApplication.wixproj]"
